init-tools.sh

```diff
@@ -157,7 +157,7 @@ if [ ! -e $__INIT_TOOLS_DONE_MARKER ]; then
     Tools-Override/crossgen.sh $__scriptpath/Tools
 
     echo "Making all .sh files executable under Tools."
-    ls $__scriptpath/Tools/*.sh | xargs chmod +x
+    find $__scriptpath/Tools/ -name "*.sh" -exec chmod +x {} \;
```

Comment:
How about:

`ls **/*sh | while read x; do chmod +x $x; done `

Seem to work on all the *nix environments I have locally.